File: src/main/java/com/paragon/client/ui/panel/Click.java
package com.paragon.client.ui.panel;

public enum Click {

    LEFT(0),
    RIGHT(1),
    MIDDLE(2),
    SIDE_ONE(3),
    SIDE_TWO(4);

    private int button;

    Click(int button) {
        this.button = button;
    }

    public int getButton() {
        return button;
    }

    public static Click getClick(int button) {
        return Click.values()[button];
    }

}

File: src/main/java/com/paragon/client/ui/panel/Click.kt
package com.paragon.client.ui.panel

/**
 * @author SooStrator1136
 */
enum class Click(val button: Int) {

    LEFT(0),
    RIGHT(1),
    MIDDLE(2),
    SIDE_ONE(3),
    SIDE_TWO(4);

    companion object {

        @JvmStatic
        fun getClick(button: Int) = values()[button]

    }

}